sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/ResourceIdentifier.cs

```diff
@@ -130,6 +133,25 @@ internal static ResourceIdentifier CreateBaseSubscriptionIdentifier(string subsc
                         }
                     default:
                         {
+                            // Check for resources that has implicit(omitted) provider
+                            if (ResourceWithImplicitProviderList.Keys.Contains(parts[0].ToLowerInvariant()))
```

Comment:
does this assume that everything in ResourceWithImplicitProviderList will always stem from a SubscriptionResourceId?  If so can we rename the list appropriately?